src/promptflow/promptflow/azure/operations/_arm_connection_operations.py

```diff
@@ -262,6 +262,9 @@ def _build_connection_dict(cls, name, subscription_id, resource_group_name, work
                 "for current workspace, and wait for a few minutes to make sure the new role takes effect. "
             )
             raise OpenURLUserAuthenticationError(message=auth_error_message)
+        except Exception as e:
+            raise UserErrorException(target=ErrorTarget.CONTROL_PLANE_SDK, message=str(e), error=e)
```

Comment:
This should be service error?

Reply:
I am referring to `OpenURLUserAuthenticationError (message=autherror message)`, which is a UserError.

Reply:
Based on our discussion, I have made modifications.
![image](https://github.com/microsoft/promptflow/assets/23182548/5f145153-470f-42f7-8aff-9c37cf5e8c7a)